sql/index_builder.go

```diff
@@ -243,7 +243,7 @@ func (b *IndexBuilder) Ranges(ctx *Context) RangeCollection {
 			b.err = err
 			return nil
 		}
-		if !isempty {
+		if !isempty || b.idx.IsSpatial() {
```

Comment:
does spatial indexes IsEmpty() not return something sensible? i'd maybe prefer the interface to be a reliable indicator of empty

Reply:
`Range.IsEmpty()` will call `RangeColumnExpr.IsEmpty()`, which checks if `UpperBound >= LowerBound`.
It's possible for the bounding box to be just a point (`UpperBound == LowerBound`), which is incorrectly classified as empty.

A better fix might be to just have a `SpatialIndexBuilder` with different logic like you mentioned before.